int32 dim
float64 field
float64[] gradient
float64[] hessian
